### PR TITLE
[5.1] Use Dialog for Module editing

### PR DESCRIPTION
**Pull request by Fedik**
### Summary of Changes

Use Dialog for module editing modals.

Affects:
- Dashboard: add module
- Menus: add/edit module
- Menu item: Module Assignment tab

Any other place I forgot?

### Testing Instructions
Run `npm install`,

Test:
- Dashboard: add module
- Menus: add/edit module
- Menu item, in Module Assignment tab: edit module

### Actual result BEFORE applying this Pull Request
Works with BS modal

### Expected result AFTER applying this Pull Request
Works with new dialog

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed
- [x] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/208
- [ ] No documentation changes for manual.joomla.org needed

References:
- https://github.com/joomla/joomla-cms/pull/40150

### CODE REVIEW

File: administrator/components/com_cpanel/tmpl/cpanel/default.php [hunk 1]
```diff
@@ -28,26 +28,16 @@
 /** @var Joomla\CMS\WebAsset\WebAssetManager $wa */
 $wa = $this->document->getWebAssetManager();
 $wa->useScript('com_cpanel.admin-cpanel')
-    ->useScript('com_cpanel.admin-addmodule');
+    ->useScript('joomla.dialog-autocreate');
 
 $user = $this->getCurrentUser();
 
-// Set up the bootstrap modal that will be used for all module editors
-echo HTMLHelper::_(
```
Comment (Quy) [inline administrator/components/com_cpanel/tmpl/cpanel/default.php:36]: Please remove use statement.

Comment (Fedik) [reply administrator/components/com_cpanel/tmpl/cpanel/default.php:36]: done

**Comment by fgsw** [comment]
I have tested this item :white_check_mark: successfully on 25f275fcf50d284b6a854360912021848096567e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42423">issues.joomla.org/tracker/joomla-cms/42423</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 25f275fcf50d284b6a854360912021848096567e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42423">issues.joomla.org/tracker/joomla-cms/42423</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/42423">issues.joomla.org/tracker/joomla-cms/42423</a>.</sub>

**Comment by LadySolveig** [comment]
Thank you @Fedik and also the supportive testers @fgsw @Quy

**Pull request merged**